VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowDeployProvider.cs

```diff
@@ -0,0 +1,204 @@
+
+        public bool IsDeploySupported
+        {
+            get
+            {
+                return true;
+
+                //  IsProjectAMeadowApp().ContinueWith(t => IsDeploySupported = t.Result);
+            }
+        }
+
+        static Hcom.IMeadowConnection connection = null;
+
+        private readonly string osVersion;
+
+        [ImportingConstructor]
+        public MeadowDeployProvider(ConfiguredProject configuredProject)
+        {
+            this.configuredProject = configuredProject;
+        }
+
+        public async Task DeployAsync(CancellationToken cancellationToken, TextWriter textWriter)
+        {
+            if (await IsProjectAMeadowApp() == false)
+            {
+                return;
+            }
+
+            Globals.DebugOrDeployInProgress = true;
+
+            await outputLogger?.ConnectTextWriter(textWriter);
+            await outputLogger.ShowBuildOutputPane();
+
+            var filename = configuredProject.UnconfiguredProject.FullPath;
+
+            var projFileContent = File.ReadAllText(filename);
+
+            if (projFileContent.Contains(MeadowSDKVersion) == false)
+            {
+                DeployFailed();
+                return;
+            }
+
+            var outputPath = await GetOutputPathAsync(filename);
+
+            if (string.IsNullOrEmpty(outputPath))
+            {
+                DeployFailed();
+                return;
+            }
+
+            if (connection != null)
+            {
+                connection.FileWriteProgress -= MeadowConnection_DeploymentProgress;
+                connection.DeviceMessageReceived -= MeadowConnection_DeviceMessageReceived;
+            }
+
+            connection = MeadowConnection.GetCurrentConnection();
+
+            connection.FileWriteProgress += MeadowConnection_DeploymentProgress;
+            connection.DeviceMessageReceived += MeadowConnection_DeviceMessageReceived;
+
+            try
+            {
+                await connection.WaitForMeadowAttach();
+
+                if (await connection.IsRuntimeEnabled() == true)
+                {
+                    await connection.RuntimeDisable();
+                }
+
+                var deviceInfo = await connection.GetDeviceInfo();
+
+                string osVersion = deviceInfo.OsVersion;
+
+                var fileManager = new FileManager(null);
+                await fileManager.Refresh();
+
+                bool includePdbs = configuredProject?.ProjectConfiguration?.Dimensions["Configuration"].Contains("Debug") ?? false;
+
+                var packageManager = new PackageManager(fileManager);
+
+                await packageManager.TrimApplication(new FileInfo(Path.Combine(outputPath, "App.dll")), osVersion, includePdbs, cancellationToken: cancellationToken);
+
+                await Task.Run(async () => await AppManager.DeployApplication(packageManager, connection, osVersion, outputPath, includePdbs, false, outputLogger, cancellationToken));
```

Comment:
Do we need a `Task.Run` around an already async call here, inside this async method? A quick test without `Task.Run` showed the debugging working.